### PR TITLE
WFLY-9530 Ensure requisite cache configurations are defined when using InfinispanRegionFactory with standalone hibernate.

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.jboss.org/browse/WFLY-9530

### CODE REVIEW

**Comment by bstansberry** [comment]
@scottmarlow Please approve.

File: jpa/hibernate4_1/src/main/java/org/jboss/as/jpa/hibernate4/HibernateSecondLevelCache.java [hunk 1]
```diff
@@ -89,37 +82,35 @@ public static void addSecondLevelCacheDependencies(Properties mutableProperties,
...
             Notification.addCacheDependencies(Classification.INFINISPAN, cacheSettings);
         }
     }
+
+    public static Set<String> findCaches(Properties properties) {
+        Set<String> caches = new HashSet<>();
+
+        caches.add(properties.getProperty(ENTITY_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
+        caches.add(properties.getProperty(COLLECTION_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
+        caches.add(properties.getProperty(COLLECTION_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
```
Comment (scottmarlow) [inline jpa/hibernate4_1/src/main/java/org/jboss/as/jpa/hibernate4/HibernateSecondLevelCache.java]: Please one of the duplicate calls to caches.add(properties.getProperty(COLLECTION_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE))

File: jpa/hibernate4_3/src/main/java/org/jboss/as/jpa/hibernate4/HibernateSecondLevelCache.java [hunk 2]
```diff
@@ -89,37 +82,35 @@ public static void addSecondLevelCacheDependencies(Properties mutableProperties,
...
             Notification.addCacheDependencies(Classification.INFINISPAN, cacheSettings);
         }
     }
+
+    public static Set<String> findCaches(Properties properties) {
+        Set<String> caches = new HashSet<>();
+
+        caches.add(properties.getProperty(ENTITY_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
+        caches.add(properties.getProperty(COLLECTION_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
+        caches.add(properties.getProperty(COLLECTION_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
```
Comment (scottmarlow) [inline jpa/hibernate4_3/src/main/java/org/jboss/as/jpa/hibernate4/HibernateSecondLevelCache.java]: Please one of the duplicate calls to caches.add(properties.getProperty(COLLECTION_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE))

File: jpa/hibernate5/src/main/java/org/jboss/as/jpa/hibernate5/HibernateSecondLevelCache.java [hunk 3]
```diff
@@ -88,41 +79,39 @@ public static void addSecondLevelCacheDependencies(Properties mutableProperties,
...
         }
     }
+
+    public static Set<String> findCaches(Properties properties) {
+        Set<String> caches = new HashSet<>();
+
+        caches.add(properties.getProperty(ENTITY_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
+        caches.add(properties.getProperty(IMMUTABLE_ENTITY_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
+        caches.add(properties.getProperty(COLLECTION_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
+        caches.add(properties.getProperty(COLLECTION_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
```
Comment (scottmarlow) [inline jpa/hibernate5/src/main/java/org/jboss/as/jpa/hibernate5/HibernateSecondLevelCache.java]: Please remove one of the duplicate calls to caches.add(properties.getProperty(COLLECTION_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE))

File: jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/secondLevelCache/InfinispanCacheDeploymentListener.java [hunk 4]
```diff
@@ -57,15 +62,8 @@
...
-    public static final String COLLECTION = "collection";
-    public static final String ENTITY = "entity";
-    public static final String IMMUTABLE_ENTITY = "immutable-entity";
     public static final String NAME = "name";
-    public static final String NATURAL_ID = "natural-id";
-    public static final String QUERY = "query";
-    public static final String TIMESTAMPS = "timestamps";
-    public static final String PENDING_PUTS = "pending-puts";
-    public static final String CUSTOM = "custom";
+    public static final String CACHES = "caches";
```
Comment (scottmarlow) [inline jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/secondLevelCache/InfinispanCacheDeploymentListener.java]: Please remind me, where is the code that implements "caches"?  Thanks!

File: jpa/hibernate4_1/src/main/java/org/jboss/as/jpa/hibernate4/HibernateSecondLevelCache.java (hunk 1 above)
Comment (pferraro) [reply jpa/hibernate4_1/src/main/java/org/jboss/as/jpa/hibernate4/HibernateSecondLevelCache.java]: Oops.  Will fix.

File: jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/secondLevelCache/InfinispanCacheDeploymentListener.java (hunk 4 above)
Comment (pferraro) [reply jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/secondLevelCache/InfinispanCacheDeploymentListener.java]: https://github.com/wildfly/wildfly/pull/10639/files/ca191c84ccfe190219edb90824a602bb71a0dca2#diff-d5db9279d2617193a1ccc79e56a0fad7R95

File: jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/secondLevelCache/InfinispanCacheDeploymentListener.java [hunk 5]
```diff
@@ -85,96 +83,88 @@ public Wrapper startCache(Classification classification, Properties properties)
...
+                    .awaitValue();
 
+            // Ensure cache configuration services are started
+            for (ServiceController<Configuration> controller : controllers) {
+                serviceNames.add(controller.getName());
+                controller.awaitValue();
+            }
         } else {
             // need a shared cache for jpa applications
-            serviceName = containerServiceName;
```
Comment (scottmarlow) [inline jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/secondLevelCache/InfinispanCacheDeploymentListener.java]: Are we purposely not adding the serviceName to serviceNames?

Comment (pferraro) [reply jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/secondLevelCache/InfinispanCacheDeploymentListener.java]: Correct.

**Comment by scottmarlow** [comment]
@pferraro @bstansberry I approve of merging this change.

**Comment by scottmarlow** [comment]
It would be good to merge this change to WildFly master and get community feedback, before merging the back ports.

**Comment by scottmarlow** [comment]
It would be good to merge this change to WildFly master and get community feedback, before merging the back ports.

**Comment by pferraro** [comment]
@bstansberry ^

**Comment by scottmarlow** [comment]
@pferraro would it be possible to merge my test enhancement to your community PR?  I made some small improvements to the test to give us more confidence that the Hibernate session is operating on the JTA transaction.

**Comment by pferraro** [comment]
@scottmarlow Done.

**Comment by bstansberry** [comment]
Thanks, @scottmarlow!

**Pull request merged**